src/Altinn.Platform/Altinn.Platform.Events/Events/Repository/EventsPostgresRepository.cs

```diff
@@ -0,0 +1,55 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Platform.Events.Configuration;
+using Altinn.Platform.Events.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Npgsql;
+
+namespace Altinn.Platform.Events.Repository
+{
+    /// <summary>
+    /// Handles events repository. 
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class EventsPostgresRepository : IEventsPostgresRepository
+    {
+        private NpgsqlConnection _conn;
+        private readonly ILogger _logger;
+        private readonly string insertEventSql = "call events.insert_event(@id, @source, @subject, @type, @cloudevent)";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventsPostgresRepository"/> class.
+        /// </summary>
+        public EventsPostgresRepository(IOptions<PostgreSQLSettings> postgresSettings, ILogger<EventsPostgresRepository> logger)
+        {
+            PostgresDatabaseHandler database = new PostgresDatabaseHandler(postgresSettings.Value);
+            _conn = database.GetConnection();
+            _logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> Create(CloudEvent cloudEvent)
+        {
```

Comment:
we should propably wrap this  in a try/catch/finally.. we should ensure that the connection is always closed regardless if the command succeeds or not

Reply:
Something like this? 

```cs
            try
            {
                await _conn.OpenAsync();

                NpgsqlCommand pgcom = new NpgsqlCommand(insertEventSql, _conn);
                pgcom.Parameters.AddWithValue("id", item.Id);
                pgcom.Parameters.AddWithValue("source", item.Source.OriginalString);
                pgcom.Parameters.AddWithValue("subject", item.Subject);
                pgcom.Parameters.AddWithValue("type", item.Type);
                pgcom.Parameters.AddWithValue("cloudevent", JsonConvert.SerializeObject(item));

                int res = await pgcom.ExecuteNonQueryAsync();
                return item.Id;
            }
            catch (Exception e)
            {
                Console.WriteLine($" EventsRepository // Create // Exception {JsonConvert.SerializeObject(e)}");
                throw e;
            }
            finally
            {
                await _conn.CloseAsync();
            }
```

Reply:
Done